### ISSUE TITLE
unused = meta_task((self.support_x[0], self.query_x[0], self.support_y[0], self.query_y[0]), False)

### ISSUE THREAD

**Issue by dragen1860** [cbfinn/maml#28]
Hi, I'm working on the source code of MAML, Thanks for your sharing.

1. when I try to understand 
```			
# to initialize the batch norm variables, might want to combine this, and not run idx 0 twice.
unused = meta_task((self.support_x[0], self.query_x[0], self.support_y[0], self.query_y[0]), False)
```
I don't know why need this line since it's unused??

2. and 
```
task_outputa = self.forward(inputa, weights, reuse=reuse)  # only reuse on the first iter
```
This seems to be the only line not reusing variable, I don't understand why this line need NOT reuse??
Thanks.

**Comment by cbfinn** [comment]
It is needed when using batch norm, to initialize the batch norm variables -- reuse is set to False here to initialize the batch norm variables that haven't been created yet. After that, the batch norm variables are reused.
The reuse flag is *only* for the batch norm variables. All other parameters are handled separately.

See the tensorflow documentation for details on how reuse works.
